components/suggest/src/store.rs

```diff
@@ -344,6 +344,42 @@ where
                         // records, since we understand it now.
                         dao.drop_unparsable_record_id(&record_id)?;
 
+                        Ok(())
+                    })?;
+                }
+                SuggestRecord::Amo => {
+                    let Some(attachment) = record.attachment.as_ref() else {
+                        // An Amo record should always have an
+                        // attachment with suggestions. If it doesn't, it's
+                        // malformed, so skip to the next record.
```

Comment:
```suggestion
                        // An Amo record should always have an
                        // attachment with suggestions. If it doesn't, it's
                        // malformed, so skip to the next record.
```

nitpick, style: I think we can drop the comments (and extra blank lines) in this new block, since they're the same as the block above.

This is totally a judgment call, and I'm wondering how you feel about it! My thinking is, because your new logic is so similar to what's just above it, and we already explain how that code works—the duplicated comments in this arm don't tell a reader anything new. If I were reading this code for the first time, I'd be thinking, "the code structure looks super similar to what I just read, but the spacing and comments here give me the feeling I should be paying closer attention".

But that's me. What do you think? 😊

You mentioned the duplication on Zoom earlier, too, and that got me thinking if it would be worth DRYing out this logic. But I don't think we're there just yet. While it is really similar to AMP-Wikipedia, the duplication isn't too bad now, and it's the kind of thing that's great to do when you're pretty sure you've found the generalization.

I _just_ saw [this post](https://programmingisterrible.com/post/176657481103/repeat-yourself-do-more-than-one-thing-and) earlier today, and thought you might like it, too! 😁 I feel like we're at the "repeat yourself to find abstractions" stage now—and the next time we go to add another suggestion type, we can see if we're following the same pattern, and DRY it out then. IME, it's usually been much easier to add abstractions later, than to undo a premature one. WDYT?